be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/ElectionHandler.scala

```diff
@@ -41,25 +42,27 @@ class ElectionHandler(dbRef: => AskableActorRef) extends MessageHandler {
   /** Overrides default DbActor with provided parameter
     */
   override final val dbActor: AskableActorRef = dbRef
+  private val serverUnexpectedAnswer: String = "The server is doing something unexpected"
 
   def handleSetupElection(rpcMessage: JsonRpcRequest): GraphMessage = {
     // FIXME: add election info to election channel/electionData
-    val message: Message = rpcMessage.getParamsMessage.get
-    val data: SetupElection = message.decodedData.get.asInstanceOf[SetupElection]
-    val electionId: Hash = data.id
-    val electionChannel: Channel = Channel(s"${rpcMessage.getParamsChannel.channel}${Channel.CHANNEL_SEPARATOR}$electionId")
-    val keyPair = KeyPair()
 
     // need to write and propagate the election message
     val combined = for {
+      (_, message, somedata) <- extractParameters[SetupElection](rpcMessage, serverUnexpectedAnswer)
+      data = somedata.get
```

Comment:
I believe you can replace somedata by Some(data) and in doing so avoid line 53. 

Beside that and the formatting (fixed by running "sbt scalafmt" in be2-scala directory), it seems all good to me.